packages/metro-transform-worker/src/index.js

```diff
@@ -382,21 +387,21 @@ async function transformJS(
     if (config.unstable_disableModuleWrapping === true) {
       wrappedAst = ast;
     } else {
+      let moduleFactoryName;
+      if (options.dev && !minify) {
+        moduleFactoryName = file.filename;
```

Comment:
My main concern here (other than the obvious: we should improve the profiling & debugging tools so that this isn't needed 😅) is around the portability and cache consistency of using this `filename` property.

I'm not sure off the top of my head whether this is (1) an absolute or relative path, (2) a platform-dependent or normalised path. Could you confirm?

* If the path is absolute, then transformer cache artifacts will be sensitive to the project's root directory, which is incorrect. (Cache artifacts should be compatible across machines, including e.g. checkouts at different base locations)
* If the path is absolute *and* platform-dependent, then it would begin with `C:\foo` on Windows but `/foo` on POSIX, which again is non-portable.

Reply:
This is a relative path in my testing — see the screenshot (e.g. `node_modules/blabla/...`).

I'm not sure if it's platform-dependent or not but we replace all non-valid JS ident characters with `_` so a path with `/` and `\` would result in the same identifier.

Reply:
Note that we already emit a path into the DEV artifacts [here](https://github.com/facebook/metro/blob/bbdd7d7c5e6e0feb50a9967ffae1f723c1d7c4e8/packages/metro/src/DeltaBundler/Serializers/helpers/js.js#L101) (the module's "verbose name") but it happens at a different stage so I'm not entirely sure if they're constructed in the same way or come from different sources.